subsys/net/lib/sockets/sockets.c

```diff
@@ -406,24 +423,28 @@ static inline ssize_t zsock_recv_stream(struct net_context *ctx,
 		/* Actually copy data to application buffer */
 		memcpy(buf, frag->data, recv_len);
 
-		if (recv_len != frag_len) {
-			net_buf_pull(frag, recv_len);
-		} else {
-			frag = net_pkt_frag_del(pkt, NULL, frag);
-			if (!frag) {
-				/* Finished processing head pkt in
-				 * the fifo. Drop it from there.
-				 */
-				k_fifo_get(&ctx->recv_q, K_NO_WAIT);
-				if (net_pkt_eof(pkt)) {
-					sock_set_eof(ctx);
+		if (!(flags & ZSOCK_MSG_PEEK)) {
+			if (recv_len != frag_len) {
+				net_buf_pull(frag, recv_len);
+			} else {
+				frag = net_pkt_frag_del(pkt, NULL, frag);
+				if (!frag) {
+					/* Finished processing head pkt in
+					 * the fifo. Drop it from there.
+					 */
+					k_fifo_get(&ctx->recv_q, K_NO_WAIT);
+					if (net_pkt_eof(pkt)) {
+						sock_set_eof(ctx);
+					}
+					net_pkt_unref(pkt);
```

Comment:
Please put empty lines after a block end "}"

Reply:
I did not really change this code block, I only put it inside a if-block.
On which line do you think I should put a empty line?

Reply:
In Zephyr coding style we put empty line after for example "if {}" block.
So in this case it should looke like:

```
if (net_pkt_eof(pkt)) {
	sock_set_eof(ctx);
}

net_pkt_unref(pkt);
```